frontend/src/hooks/useLocationFilter.ts

```diff
@@ -0,0 +1,34 @@
+import {useCallback, useState} from "react";
+import {FilterController, UseFilterHook} from "@/types/filtering.types";
+
+export interface LocationFilterController extends FilterController {
+  setSearchText: (searchText: string) => void;
+  searchText: string;
+}
+
+const queryDefinitions = [
+  "filters[$and][0][location][$or][0][City][$containsi]",
+  "filters[$and][0][location][$or][1][Name][$containsi]",
+  "filters[$and][0][location][$or][2][Street][$containsi]",
+  "filters[$and][0][location][$or][3][Zip][$containsi]"
+];
+
+export const useLocationFilter: UseFilterHook<LocationFilterController> = (filters, setFilters) => {
+  const [searchText, _setSearchText] = useState<string>('');
+
+  const setSearchText = useCallback((value: string) => {
```

Comment:
this is the same as in the CatergoryFilter right? perhaps we could make a shared function for this?